test/components/structures/TimelinePanel-test.js

```diff
@@ -321,31 +332,22 @@ describe('TimelinePanel', function() {
             expect(messagePanel.props.suppressFirstDateSeparator).toBe(false);
             var events = scryEventTiles(panel);
             expect(events[0].props.mxEvent).toBe(timeline.getEvents()[0]);
-            expect(events.length).toBeLessThanOrEqualTo(TIMELINE_CAP);
-
-            // we should now be able to scroll down, and paginate in the other
-            // direction.
-            setScrollTop(scrollingDiv.scrollHeight);
-            scrollingDiv.scrollTop = scrollingDiv.scrollHeight;
-
-            // the delay() below is a heinous hack to deal with the fact that,
-            // without it, we may or may not get control back before the
-            // forward pagination completes. The delay means that it should
-            // have completed.
-            return awaitScroll().delay(0);
+
+            // Expect to be able to paginate forwards, having unpaginated a few events
+            expect(panel.state.canForwardPaginate).toBe(true);
+
+            // scroll all the way to the bottom
+            return scrollDown(3);
```

Comment:
how about scrolling down until we see the last event, rather than a fixed number of pagination lumps?